modules/custom/az_publication/az_publication.install

```diff
@@ -83,3 +86,159 @@ function az_publication_update_1020702() {
+}
+
+/**
+ * Map and move data from static text field to dynamic type reference field.
+ */
+function az_publication_update_1020902(&$sandbox) {
+  // Initialize or retrieve the batch progress information.
+  if (!isset($sandbox['progress'])) {
+    $sandbox['progress'] = 0;
+    $sandbox['current_node'] = 0;
+
+    // Query all node IDs of type 'az_publication'.
+    $sandbox['nids'] = \Drupal::entityQuery('node')
+      ->condition('type', 'az_publication')
+      ->accessCheck(FALSE)
+      ->execute();
+
+    $sandbox['max'] = count($sandbox['nids']);
+  }
+
+  // Define the mapping from old field values to new config entity IDs.
+  $mapping = [
+    'article' => 'az_article',
+    'article-journal' => 'az_article_journal',
+    'article-magazine' => 'az_article_magazine',
+    'article-newspaper' => 'az_article_newspaper',
+    'bill' => 'az_bill',
+    'book' => 'az_book',
+    'broadcast' => 'az_broadcast',
+    'chapter' => 'az_chapter',
+    'classic' => 'az_classic',
+    'collection' => 'az_collection',
+    'dataset' => 'az_dataset',
+    'document' => 'az_document',
+    'entry' => 'az_entry',
+    'entry-dictionary' => 'az_entry_dictionary',
+    'entry-encyclopedia' => 'az_entry_encyclopedia',
+    'event' => 'az_event',
+    'figure' => 'az_figure',
+    'graphic' => 'az_graphic',
+    'hearing' => 'az_hearing',
+    'interview' => 'az_interview',
+    'legal_case' => 'az_legal_case',
+    'legislation' => 'az_legislation',
+    'manuscript' => 'az_manuscript',
+    'map' => 'az_map',
+    'motion_picture' => 'az_motion_picture',
+    'musical_score' => 'az_musical_score',
+    'pamphlet' => 'az_pamphlet',
+    'paper-conference' => 'az_paper_conference',
+    'patent' => 'az_patent',
+    'performance' => 'az_performance',
+    'periodical' => 'az_periodical',
+    'personal_communication' => 'az_personal_communication',
+    'post' => 'az_post',
+    'post-weblog' => 'az_post_weblog',
+    'regulation' => 'az_regulation',
+    'report' => 'az_report',
+    'review' => 'az_review',
+    'review-book' => 'az_review_book',
+    'software' => 'az_software',
+    'song' => 'az_song',
+    'speech' => 'az_speech',
+    'standard' => 'az_standard',
+    'thesis' => 'az_thesis',
+    'treaty' => 'az_treaty',
+    'webpage' => 'az_webpage',
+  ];
+
+  // Process nodes in batches.
+  // Number of nodes to process per batch.
+  $limit = 50;
+  $nids_slice = array_slice($sandbox['nids'], $sandbox['progress'], $limit);
+  $nodes = Node::loadMultiple($nids_slice);
+
+  foreach ($nodes as $node) {
+    if ($node->hasField('field_az_publication_type') && !$node->get('field_az_publication_type')->isEmpty()) {
+      $old_value = $node->get('field_az_publication_type')->value;
+
+      // Map and update the new entity reference field.
+      if (isset($mapping[$old_value])) {
+        $node->set('field_az_publication_mapped_type', ['target_id' => $mapping[$old_value]]);
+        $node->save();
```

Comment:
This method affects the updated date of publications which, might not be what we want.

Reply:
Possibly add revision and revision note mentioning why this was updated.